parser/fms_yaml_output.F90

```diff
@@ -132,13 +132,14 @@ subroutine write_yaml_from_struct_3 (yamlname, a1size, keys, vals, a2size, key2,
 type (fmsYamlOutValues_type) :: val3(a3size) !< Values corresponding to keys2
 integer (c_int)        :: lvl2keyeach(lvl2_key_parameter) !< amount of key2 'blocks' to print per level2key in keys
 end subroutine write_yaml_from_struct_3
-!> adds a level 2 key to the list.
+!> adds a level 2 key (starts new tabbed section) to the list.
 !! Will print level 2 keys in the order added.
-subroutine yaml_out_add_level2key(key_name, keytype) bind(C, name="add_level2key")
+subroutine yaml_out_add_level2key(key_length, key_name, keytype) bind(C, name="add_level2key")
```

Comment:
don't the lengths have to be static and the same between fortran and C?

Reply:
yeah i added it in to deal with gcc/CI issues. Intel will add the null char on fortran strings passed to c, gcc will not.

Reply:
You should always append the c_null_char to the end of c strings in fortran.
```fotran
"string"//c_null_char
```